### PR TITLE
Add overridable tags

### PR DESCRIPTION
**Pull request by SarahW91**
### Description

Added additional Overridable tags for a more finegrained control over React components.

### Checklist

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Comment by SarahW91** [comment]
We still need a decision on this.

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js [hunk 1]
```diff
@@ -53,67 +54,77 @@ export class DatesField extends Component {
...
-                  type="button"
-                >
-                  <Icon name="close" />
-                </Button>
-              </Form.Field>
-            </GroupField>
-          );
-        }}
-      </ArrayField>
+      <Overridable id="InvenioRdmRecords.DatesField.AddDate.ArrayField">
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js:57]:
```suggestion
      <Overridable id="InvenioRdmRecords.DatesField.AddDate.container">
```

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunityHeader/CommunityHeader.js [hunk 2]
```diff
@@ -64,47 +66,53 @@ class CommunityHeaderComponent extends Component {
...
+                                : i18next.t("Select a community")
+                            }
+                          />
+                        </Overridable>
+                      }
                     />
-                  )}
-                </>
+                    {community && (
+                      <Overridable id="InvenioRdmRecords.CommunityHeader.RemoveCommunity.Button">
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunityHeader/CommunityHeader.js:101]:
```suggestion
           <Overridable id="InvenioRdmRecords.CommunityHeader.RemoveCommunity.container" community={community}>
                    {community && (

```
passing the community as a prop to the overridable component will allow to control also the conditional statement on when this component is displayed (while overriding)

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunityHeader/CommunityHeader.js [hunk 3]
```diff
@@ -64,47 +66,53 @@ class CommunityHeaderComponent extends Component {
...
+                        changeSelectedCommunity(community);
+                        this.setState({ modalOpen: false });
+                      }}
+                      onModalChange={(value) => this.setState({ modalOpen: value })}
+                      modalOpen={modalOpen}
+                      chosenCommunity={community}
+                      displaySelected
+                      record={record}
+                      trigger={
+                        <Overridable id="InvenioRdmRecords.CommunityHeader.SelectCommunity.Button">
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunityHeader/CommunityHeader.js:82]: in the ID creation we try to follow the hierarchy and the real names of the components involved, to help identify them better (and also to help the developer to find them easier while overriding) so we try to include the actual name of the component we are overriding, therefore I would suggest to have:
```suggestion
                        <Overridable id="InvenioRdmRecords.CommunityHeader.CommunitySelectionModal.trigger">
```

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js [hunk 4]
```diff
@@ -53,67 +54,77 @@ export class DatesField extends Component {
...
+        >
+          {({ array, arrayHelpers, indexPath, value }) => {
+            const fieldPathPrefix = `${fieldPath}.${indexPath}`;
+            const requiredOption = this.getRequiredOption(value, array);
+            const hasRequiredDateValue = _has(requiredOption, "date");
+            const hasRequiredTypeValue = _has(requiredOption, "type");
+            const hasRequiredDescriptionValue = _has(requiredOption, "description");
+            return (
+              <GroupField fieldPath={fieldPath} optimized>
+                <Overridable id="InvenioRdmRecords.DatesField.Date.TextField">
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js:79]:
```suggestion
                <Overridable id="InvenioRdmRecords.DatesField.Date.container">
```

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js [hunk 5]
```diff
@@ -53,67 +54,77 @@ export class DatesField extends Component {
...
+                  <TextField
+                    fieldPath={`${fieldPathPrefix}.date`}
+                    label={i18next.t("Date")}
+                    placeholder={placeholderDate}
+                    disabled={hasRequiredDateValue}
+                    required
+                    width={5}
+                  />
+                </Overridable>
+                <Overridable id="InvenioRdmRecords.DatesField.Type.SelectField">
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js:89]:
```suggestion
                <Overridable id="InvenioRdmRecords.DatesField.Type.container">
```

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js [hunk 6]
```diff
@@ -53,67 +54,77 @@ export class DatesField extends Component {
...
+                    label={i18next.t("Type")}
+                    aria-label={i18next.t("Type")}
+                    options={sortOptions(options.type)}
+                    disabled={hasRequiredTypeValue}
+                    required
+                    width={5}
+                    optimized
+                  />
+                </Overridable>
+                <Overridable id="InvenioRdmRecords.DatesField.Description.TextField">
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js:101]:
```suggestion
                <Overridable id="InvenioRdmRecords.DatesField.Description.container">
```

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js [hunk 7]
```diff
@@ -53,67 +54,77 @@ export class DatesField extends Component {
...
+                  <TextField
+                    fieldPath={`${fieldPathPrefix}.description`}
+                    label={i18next.t("Description")}
+                    disabled={hasRequiredDescriptionValue}
+                    width={5}
+                  />
+                </Overridable>
+                <Overridable id="InvenioRdmRecords.DatesField.Remove.Button">
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js:109]:
```suggestion
                <Overridable id="InvenioRdmRecords.DatesField.Remove.container">
```

**Review by kpsherva — COMMENTED** [review]
hello @SarahW91 ! my apologies for sizeable comments, I have a feeling that maybe discussion over a zoom call would help us finding a better alignment on what you are trying to achieve as an end result ? I am worried that maybe I am missing the context of your changes and I could advise you a bit better if I understand exactly on how you want to customize the form. What do you think?

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js [hunk 8]
```diff
@@ -53,67 +54,77 @@ export class DatesField extends Component {
...
-                  type="button"
-                >
-                  <Icon name="close" />
-                </Button>
-              </Form.Field>
-            </GroupField>
-          );
-        }}
-      </ArrayField>
+      <Overridable id="InvenioRdmRecords.DatesField.AddDateArrayField.Container">
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js:57]:
```suggestion
      <Overridable id="InvenioRdmRecords.DatesField.AddDateArrayField.Container" fieldPath={fieldPath}>
```

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js [hunk 9]
```diff
@@ -53,67 +54,77 @@ export class DatesField extends Component {
...
+        >
+          {({ array, arrayHelpers, indexPath, value }) => {
+            const fieldPathPrefix = `${fieldPath}.${indexPath}`;
+            const requiredOption = this.getRequiredOption(value, array);
+            const hasRequiredDateValue = _has(requiredOption, "date");
+            const hasRequiredTypeValue = _has(requiredOption, "type");
+            const hasRequiredDescriptionValue = _has(requiredOption, "description");
+            return (
+              <GroupField fieldPath={fieldPath} optimized>
+                <Overridable id="InvenioRdmRecords.DatesField.DateTextField.Container">
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/DatesField/DatesField.js:79]:
```suggestion
                <Overridable id="InvenioRdmRecords.DatesField.DateTextField.Container" fieldPath={`${fieldPathPrefix}.date`}>
```
please also add to the occurences below

**Pull request merged**